BeeKit/Model/DueBy/BeeminderDueByEntry.swift

```diff
@@ -0,0 +1,56 @@
+// Part of BeeSwift. Copyright Beeminder
+
+import SwiftyJSON
+
+public class BeeminderDueByEntry: NSObject, NSSecureCoding, Codable {
+    public let total: Double
+    public let delta: Double
+    public let formatted_total_for_beedroid: String
+    public let formatted_delta_for_beedroid: String
+    
+    public init(json: JSON) {
+        self.delta = json["delta"].doubleValue
+        self.total = json["total"].doubleValue
+        
+        self.formatted_delta_for_beedroid = json["formatted_delta_for_beedroid"].stringValue
+        self.formatted_total_for_beedroid = json["formatted_total_for_beedroid"].stringValue
+    }
+    
+    init(total: Double, delta: Double, formatted_total_for_beedroid: String, formatted_delta_for_beedroid: String) {
+        self.total = total
+        self.delta = delta
+        self.formatted_total_for_beedroid = formatted_total_for_beedroid
+        self.formatted_delta_for_beedroid = formatted_delta_for_beedroid
+    }
+    
+    public static var supportsSecureCoding: Bool { true }
+
+    private enum Key: String {
+        case total
+        case delta
+        case formattedTotalForBeedroid
+        case formattedDeltaForBeedroid
+    }
+    
+    public func encode(with coder: NSCoder) {
```

Comment:
Do we need all this custom encode/decode logic? My understanding is Swift (4.0+) should be able to synthesize it.

Reply:
Codable can take care of some of it. The rest is for NSCoding (NSSecureCoding). We can make one use the other, though this is not available out of the box, afaik. Will look into it

Reply:
If we are using NSSecureCoding, can we use `NSSecureUnarchiveFromDataTransformer` instead of a custom one?

Reply:
I do not think so. I think if custom, core data non-native, are being stored, then a custom valuetransformer is needed.

Reply:
> Do we need all this custom encode/decode logic?

This uses the pattern the rest of the app uses: SwiftyJSON and manually plucking out from the json.

> My understanding is Swift (4.0+) should be able to synthesize it.

NSCoding's encode: `public func encode(with coder: NSCoder)`
Codable's encode: `public func encode(to encoder: Encoder) throws`